lib/src/facts/aix/kernel_resolver.cc

```diff
@@ -0,0 +1,25 @@
+#include <internal/facts/aix/kernel_resolver.hpp>
+#include <facter/facts/collection.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <sys/utsname.h>
+
+using namespace std;
+
+namespace facter { namespace facts { namespace aix {
+
+    kernel_resolver::data kernel_resolver::collect_data(collection& facts)
+    {
+        data result;
+        struct utsname name;
+        if (uname(&name) == -1) {
+            LOG_WARNING("uname failed: %1% (%2%): kernel facts are unavailable.", strerror(errno), errno);
+            return result;
+        }
+
+        result.name = name.sysname;
+        result.release = name.release;
+        result.version = name.version;
```

Comment:
A one line `#ifdef` is fine by me; it doesn't sound like that would grow into #ifdef hell.